central/postgres/schema/schema_test.go

```diff
@@ -0,0 +1,389 @@
+			name:        WatchedImagesTableName,
+			createStmts: CreateTableWatchedImagesStmt,
+		},
+		{
+			name:        TestSingleKeyStructsTableName,
+			createStmts: CreateTableTestSingleKeyStructsStmt,
+		},
+
+		{
+			name:        TestChild1TableName,
+			createStmts: CreateTableTestChild1Stmt,
+		},
+		{
+			name:        TestGrandparentsTableName,
+			createStmts: CreateTableTestGrandparentsStmt,
+		},
+		{
+			name:        TestParent1TableName,
+			createStmts: CreateTableTestParent1Stmt,
+		},
+
+		{
+			name:        TestParent3TableName,
+			createStmts: CreateTableTestParent3Stmt,
+		},
+		{
+			name:        TestParent2TableName,
+			createStmts: CreateTableTestParent2Stmt,
+		},
+		{
+			name:        TestGrandChild1TableName,
+			createStmts: CreateTableTestGrandChild1Stmt,
+		},
+		{
+			name:        TestG2GrandChild1TableName,
+			createStmts: CreateTableTestG2GrandChild1Stmt,
+		},
+		{
+			name:        TestG3GrandChild1TableName,
+			createStmts: CreateTableTestG3GrandChild1Stmt,
+		},
+		{
+			name:        TestGGrandChild1TableName,
+			createStmts: CreateTableTestGGrandChild1Stmt,
+		},
+		{
+			name:        TestChild2TableName,
+			createStmts: CreateTableTestChild2Stmt,
+		},
+	}
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.Require().Contains(allTestCases, testCase.name)
+			allTestCases.Remove(testCase.name)
+			schema := globaldb.GetSchemaForTable(testCase.name)
+			gormSchemas := s.getGormTableSchemas(schema, testCase.createStmts)
+			pgutils.CreateTable(s.ctx, s.pool, testCase.createStmts)
+			for table, gormSchema := range gormSchemas {
+				sqlSchema := s.dumpSchema(table)
+				s.Require().Equal(sqlSchema, gormSchema)
+			}
+			// Check if the table name is reversible.
+			// Gorm may have wrong behavior if the table name is not reversible.
+			schemaName := pgutils.NamingStrategy.SchemaName(testCase.name)
+			s.Require().Equal(testCase.name, pgutils.NamingStrategy.TableName(schemaName))
+		})
+	}
+	s.Require().Len(allTestCases, 0)
+}
+
+func (s *SchemaTestSuite) getAllTestCases() []string {
+	files, err := os.ReadDir(".")
+	s.Require().NoError(err)
+	var testCases []string
+	for _, file := range files {
+		name := file.Name()
+		if excludeFiles.Contains(name) || !strings.HasSuffix(name, ".go") {
+			fmt.Printf("Skipping %s\n", name)
+			continue
+		}
+		testCases = append(testCases, strings.TrimSuffix(name, ".go"))
+	}
+	return testCases
+}
+
+func (s *SchemaTestSuite) getGormTableSchemas(schema *walker.Schema, createStmt *pkgPostgres.CreateStmts) map[string]string {
+	pgutils.CreateTableFromModel(s.gorm, createStmt)
+	defer s.dropTableFromModel(createStmt)
+	tables := s.tablesForSchema(schema)
+
+	tableMap := make(map[string]string, len(tables))
+	for _, tbl := range tables {
+		tableMap[tbl] = s.dumpSchema(tbl)
+	}
+	return tableMap
+}
+
+func (s *SchemaTestSuite) dumpSchema(table string) string {
+	// Dump Postgres schema
+	cmd := exec.Command(`pg_dump`, `--schema-only`,
```

Comment:
I think we should document requirement for `pg_dump` and add it to https://github.com/stackrox/stackrox-env/blob/master/flake.nix#L25